### PR TITLE
2024 edition update and other minor refactors

### PR DESCRIPTION
**Pull request by Coca162**
Some cleanup work for the codebase:
- Using edition 2024 for all crates
- Fixed all checks/clippy warnings
- [`clippy::wildcard_imports`](https://rust-lang.github.io/rust-clippy/stable/index.html#wildcard_imports) now warns for libnpins
- `anyhow::Result` is now always fully qualified in libnpins
- [Formatting code with the new edition](https://doc.rust-lang.org/edition-guide/rust-2024/rustfmt-formatting-fixes.html)
- `cargo update`
-  Turn off default features for `nix-compat` and set exact revision

### CODE REVIEW

**Comment by Coca162** [comment]
Seems like a spurious CI failure(?)

**Comment by Coca162** [comment]
Switched to a single threaded runtime for tokio, shouldn't have any impact but just minor efficiency gains

File: libnpins/src/lib.rs [hunk 1]
```diff
@@ -178,7 +177,7 @@ macro_rules! mkPin {
             })*
 
             /* If an error is returned, `self` remains unchanged */
-            pub async fn update(&mut self) -> Result<Vec<diff::DiffEntry>> {
+            pub async fn update(&mut self) -> ::anyhow::Result<Vec<diff::DiffEntry>> {
```
[the remark this answers is not in the document]
Comment (Coca162) [reply libnpins/src/lib.rs]: Its somewhat overkill but leading `::` use the [global root](https://doc.rust-lang.org/reference/paths.html#r-paths.qualifiers.global-root) instead of the local paths, so macro users can have a `use something::else as anyhow` without the macro suddenly breaking

**Comment by Coca162** [comment]
Since it's a relatively minor change code wise anyhow, I'll spin the single threaded tokio runtime off into another PR, I'll keep testing it in the meantime to see if I get any weird edge cases with my use.

**Comment by Coca162** [comment]
Added back the comment about clap's version that was accidentally stripped out with #186, I assume it still applies so

**Comment by Coca162** [comment]
Not sure about why the CI is failing on snix now, locally I can run it fine with this hash...

**Comment by andir** [comment]
> Not sure about why the CI is failing on snix now, locally I can run it fine with this hash...

IIRC snix has some store paths in their files and those will be reference scanned and then nix complains about undeclared / impure references. I'm not sure why that doesn't surface for you locally, but I can reproduce it on my machines.

Perhaps it is best to just delay the snix package upgrade, if we can.
